src/main/java/uk/gov/ch/controller/update/OverseasEntityDataController.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.ch.controller.update;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.ch.exception.EntityEmailNotFoundException;
+import uk.gov.ch.exception.InvalidEntityNumberException;
+import uk.gov.ch.model.update.OverseasEntityDataJson;
+import uk.gov.ch.service.update.impl.OverseasOverseasEntityDataServiceImpl;
+import uk.gov.ch.util.ApiLogger;
+
+import javax.validation.ConstraintViolationException;
+import javax.validation.Valid;
+import javax.validation.constraints.Pattern;
+import java.util.HashMap;
+
+import static uk.gov.ch.util.Constants.OVERSEAS_ENTITY_ID_KEY;
+
+@RestController
+public class OverseasEntityDataController {
+    @Autowired
+    private OverseasOverseasEntityDataServiceImpl entityDataService;
+
+    @GetMapping("/overseas-entity/{incorporationNumber}/entityData")
+    public ResponseEntity<OverseasEntityDataJson> getEntityEmail(@PathVariable @Valid @Pattern(regexp = "^OE\\d{6}$") String incorporationNumber) {
+
+        HashMap<String, Object> logMap = new HashMap<>();
+        logMap.put(OVERSEAS_ENTITY_ID_KEY, (Object) incorporationNumber);
```

Comment:
I'm wondering why you've cast the `incorporationNumber` to an Object here? This might actually cause problems for log stack as we use the same log map key in the OE API but write it as a String. Logstack (and Kibana) can then have issues with the varying datatypes...

This might actually be a good point to use the newer version of the structured logging library which provides strongly typed log map keys - see https://github.com/companieshouse/structured-logging/commit/e7eda2dce309dce7e7fa62032557ff96e9e91c16. Perhaps then the existing 'company_number' key could just be used for the log output. 

Worth a conversation with @sroberts2 I think.

Reply:
I think the map is String, Object so the cast is not needed (but happens anyway).

Reply:
Thanks for migrating to the new version of structured logging and implementing this change.